### PR TITLE
Some update for vxWorks

### PR DESCRIPTION
**Pull request by BaoshanPang**
This is to address issues:
rust-lang#1469 and rust-random/getrandom#86 (comment)

   
What have been changed in this PR:

 1. adding randBytes(), randABytes(), randUBytes(), randSecure() and taskDelay()
 2. change armv7-wrs-vxworks to armv7-wrs-vxworks-eabihf
 3. code cleanup

### CODE REVIEW

**Comment by n-salim** [comment]
Adding to @BaoshanPang 's overview of the changes:

4. Fix padding in definition of sockaddr_storage struct

**Comment by n-salim** [comment]
OK I think we kicked this off a tad too early. The rust-lang.libc test failed because we haven't yet picked up our compiler changes from

https://github.com/rust-lang/rust/pull/63604

File: src/vxworks/mod.rs [hunk 1]
```diff
@@ -1323,10 +1243,7 @@ extern {
         fd: ::c_int,
         offset: off_t,
     ) -> *mut ::c_void;
-    #[cfg_attr(
-        all(target_os = "macos", target_arch = "x86"),
-        link_name = "munmap$UNIX2003"
-    )]
+
```
Comment (gnzlbg) [inline src/vxworks/mod.rs]: spurious extra line ?

File: src/vxworks/mod.rs [hunk 2]
```diff
@@ -1391,67 +1289,28 @@ extern {
...
-    pub fn mktime(tm: *mut tm) -> time_t;
-    #[cfg_attr(target_os = "netbsd", link_name = "__time50")]
-    pub fn time(time: *mut time_t) -> time_t;
-    #[cfg_attr(target_os = "netbsd", link_name = "__gmtime50")]
-    pub fn gmtime(time_p: *const time_t) -> *mut tm;
-    #[cfg_attr(target_os = "netbsd", link_name = "__locatime50")]
-    pub fn localtime(time_p: *const time_t) -> *mut tm;
-     */
-    #[cfg_attr(target_os = "netbsd", link_name = "__difftime50")]
+
```
Comment (gnzlbg) [inline src/vxworks/mod.rs]: spurious extra line ?

**Comment by BaoshanPang** [comment]
The error here seems nothing to do with my change:

https://github.com/rust-lang/libc/pull/1473/checks?check_run_id=196203151

**Comment by BaoshanPang** [comment]
@gnzlbg 

Is this a known issue? I see the test is still failing on the same issue.

**Comment by BaoshanPang** [comment]
> @gnzlbg
> 
> Is this a known issue? I see the test is still failing on the same issue.

It looks like we are using newer version bootstrap compiler which have more strict warning/error level.

**Comment by BaoshanPang** [comment]
@gnzlbg 
Thanks!

**Comment by newpavlov** [comment]
@gnzlbg 
Any ETA on when this PR will be published on crates.io? We would like to use it for rust-random/getrandom#86.

**Comment by gnzlbg** [comment]
Probably some time after https://github.com/rust-lang/rust/pull/63806 lands.

**Pull request merged**